src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -95,17 +95,33 @@ private void AddResidualTrivia(SyntaxTriviaList trivia, bool requiresNewLine = f
             {
                 if (requiresNewLine)
                 {
-                    this.AddEndOfLine();
+                    this.AddEndOfLine(trivia);
                 }
 
                 _residualTrivia.Add(trivia);
             }
+            
+            private void AddEndOfLine(params SyntaxTriviaList[] lists)
+            {
+                if ((_options & SyntaxRemoveOptions.UseOriginalEndOfLine) != 0)
+                {
+                    // We will be forced to use the default end of line when removing the last node
+                    // in the document with KeepEndOfLine option, where SingleLineCommentTrivia is located
+                    // inside the node trailing trivia.
+                    // e.g. Removing class B from "class A {} class B {} // comment"
+                    this.AddEndOfLine(GetEndOfLine(lists) ?? SyntaxFactory.CarriageReturnLineFeed);
```

Comment:
basically... why not *always* do this?

Reply:
I've done this way in order to avoid a breaking change when `KeepEndOfLine` option is used. I would be fine with changing the behavior of `KeepEndOfLine`, so that the added EOL would be different based on the EOL used in the document. By doing so, we could get rid of the `UseOriginalEndOfLine` option added in this PR.

Reply:
> so that the added EOL would be different based on the EOL used in the document

That would be my preference.  I'm struggling to think of a user that would not want that behavior.